lib/email/receiver.rb

```diff
@@ -900,15 +902,33 @@ def add_attachments(raw, user_id, options = {})
             else
               raw << "\n\n#{attachment_markdown(upload)}\n\n"
             end
+          else
+            is_attachment_rejected = true
+            rejected_attachments << upload.original_filename
+            raw << "\n\n#{I18n.t('emails.incoming.missing_attachment', filename: upload.original_filename)}\n\n"
           end
         ensure
           tmp&.close!
         end
       end
+      notify_about_rejected_attachment(rejected_attachments) if is_attachment_rejected
```

Comment:
You don't need the `is_attachment_rejected` since you can test for `rejected_attachments.present?` instead ;)